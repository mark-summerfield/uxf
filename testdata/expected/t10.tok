TClassBegin ttype=<Files>
  Field Str("Kind")
  Field Str("Filename")
TClassEnd
TClassBegin ttype=<Categories>
  Field Str("CID")
  Field Str("Title")
  Field Str("Selected")
TClassEnd
TClassBegin ttype=<Playlists>
  Field Str("PID")
  Field Str("Title")
  Field Str("CID")
  Field Str("Selected")
TClassEnd
MapBegin
  Str("General")
  MapBegin
    Str("saved")
    Date(2022-03-21)
    Str("autosave")
    Bool(true)
    Str("historysize")
    Int(-35)
    Str("volume")
    Real(0.7)
    Str("files")
    TableBegin ttype=<Files>
      Str("current")
      Str("/home/mark/app/rs/tlm/PlaylistsTest.tlm")
      Str("recent1")
      Str("/home/mark/app/rs/tlm/PlaylistsTest.tlm")
      Str("recent2")
      Str("/home/mark/data/playlists-all.tlm")
    TableEnd
  MapEnd
  Str("Window")
  MapBegin
    Str("x")
    Int(383)
    Str("y")
    Int(124)
    Str("width")
    Int(590)
    Str("height")
    Int(536)
    Str("scale")
    Real(1.1)
  MapEnd
  Str("Magic")
  Bytes([31, 139])
  Str("Nested Map")
  MapBegin
    Str("Classical")
    ListBegin
      Int(5)
      Bool(true)
    ListEnd
    Str("Modern Instrumental")
    ListBegin
      Int(4)
      Bool(false)
    ListEnd
    Str("New Acquistions")
    ListBegin
      Int(1)
      Bool(false)
    ListEnd
    Str("Nested List of Lists")
    ListBegin
      ListBegin
        Int(5)
        Str("Classical")
        Bool(true)
      ListEnd
      ListBegin
        Int(4)
        Str("Modern Instrumental")
        Bool(false)
      ListEnd
      ListBegin
        Int(1)
        Str("New Acquistions")
        Bool(false)
      ListEnd
    ListEnd
    Str("Nested Table")
    TableBegin ttype=<Categories>
      Int(5)
      Str("Classical")
      Bool(true)
      Int(4)
      Str("Modern Instrumental")
      Bool(false)
      Int(1)
      Str("New Acquistions")
      Bool(false)
      Int(2)
      Str("Pop")
      Bool(false)
      Int(3)
      Str("Punk")
      Bool(false)
      Int(7)
      Str("Uncategorized")
      Bool(false)
      Int(6)
      Str("Unpopular Pop")
      Bool(false)
    TableEnd
    Str("Nested List of Tables")
    ListBegin
      TableBegin ttype=<Categories>
        Int(5)
        Str("Classical")
        Bool(true)
        Int(4)
        Str("Modern Instrumental")
        Bool(false)
        Int(1)
        Str("New Acquistions")
        Bool(false)
        Int(2)
        Str("Pop")
        Bool(false)
        Int(3)
        Str("Punk")
        Bool(false)
        Int(7)
        Str("Uncategorized")
        Bool(false)
        Int(6)
        Str("Unpopular Pop")
        Bool(false)
      TableEnd
      TableBegin ttype=<Playlists>
        Int(4)
        Str("ABBA")
        Int(2)
        Bool(false)
        Int(38)
        Str("Bach")
        Int(5)
        Bool(false)
        Int(39)
        Str("Bartok")
        Int(5)
        Bool(false)
        Int(5)
        Str("Beatles")
        Int(2)
        Bool(false)
        Int(40)
        Str("Beethoven")
        Int(5)
        Bool(false)
        Int(6)
        Str("Blondie")
        Int(2)
        Bool(false)
        Int(52)
        Str("Bob Marley")
        Int(6)
        Bool(true)
        Int(7)
        Str("Bruce Springsteen")
        Int(2)
        Bool(false)
        Int(41)
        Str("Chopin")
        Int(5)
        Bool(true)
        Int(37)
        Str("Classical")
        Int(5)
        Bool(false)
        Int(8)
        Str("David Bowie")
        Int(2)
        Bool(false)
        Int(9)
        Str("Dire Straits")
        Int(2)
        Bool(false)
      TableEnd
    ListEnd
  MapEnd
MapEnd
Eof
----------------------------------------
