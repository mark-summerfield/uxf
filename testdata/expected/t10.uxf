uxf 1.0 TLM Config
=Categories CID Title Selected
=Files Kind Filename
=Playlists PID Title CID Selected
{
  <General>
    {
      <saved> 2022-03-21
      <autosave> yes
      <historysize> -35
      <volume> 0.7
      <files>
        (Files
          <current> </home/mark/app/rs/tlm/PlaylistsTest.tlm>
          <recent1> </home/mark/app/rs/tlm/PlaylistsTest.tlm>
          <recent2> </home/mark/data/playlists-all.tlm>
        )
    }
  <Window>
    {<x> 383 <y> 124 <width> 590 <height> 536 <scale> 1.1}
  <Magic> (:1F8B:)
  <Nested Map>
    {
      <Classical>
        [5 yes]
      <Modern Instrumental>
        [4 no]
      <New Acquistions>
        [1 no]
      <Nested List of Lists>
        [
          [5 <Classical> yes] 
          [4 <Modern Instrumental> no] 
          [1 <New Acquistions> no]
        ]
      <Nested Table>
        (Categories
          5 <Classical> yes
          4 <Modern Instrumental> no
          1 <New Acquistions> no
          2 <Pop> no
          3 <Punk> no
          7 <Uncategorized> no
          6 <Unpopular Pop> no
        )
      <Nested List of Tables>
        [
          (Categories
            5 <Classical> yes
            4 <Modern Instrumental> no
            1 <New Acquistions> no
            2 <Pop> no
            3 <Punk> no
            7 <Uncategorized> no
            6 <Unpopular Pop> no
          ) 
          (Playlists
            4 <ABBA> 2 no
            38 <Bach> 5 no
            39 <Bartok> 5 no
            5 <Beatles> 2 no
            40 <Beethoven> 5 no
            6 <Blondie> 2 no
            52 <Bob Marley> 6 yes
            7 <Bruce Springsteen> 2 no
            41 <Chopin> 5 yes
            37 <Classical> 5 no
            8 <David Bowie> 2 no
            9 <Dire Straits> 2 no
          )
        ]
    }
}
