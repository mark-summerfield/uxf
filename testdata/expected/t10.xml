<?xml version="1.0" ?>
<uxf version="1.0" custom="TLM Config">
  <ttypes>
    <ttype name="Categories">
      <field name="CID"/>
      <field name="Title"/>
      <field name="Selected"/>
    </ttype>
    <ttype name="Files">
      <field name="Kind"/>
      <field name="Filename"/>
    </ttype>
    <ttype name="Playlists">
      <field name="PID"/>
      <field name="Title"/>
      <field name="CID"/>
      <field name="Selected"/>
    </ttype>
  </ttypes>
  <map>
    <key>
      <str>General</str>
    </key>
    <value>
      <map>
        <key>
          <str>autosave</str>
        </key>
        <value>
          <yes/>
        </value>
        <key>
          <str>files</str>
        </key>
        <value>
          <table name="Files">
            <row>
              <str>current</str>
              <str>/home/mark/app/rs/tlm/PlaylistsTest.tlm</str>
            </row>
            <row>
              <str>recent1</str>
              <str>/home/mark/app/rs/tlm/PlaylistsTest.tlm</str>
            </row>
            <row>
              <str>recent2</str>
              <str>/home/mark/data/playlists-all.tlm</str>
            </row>
          </table>
        </value>
        <key>
          <str>historysize</str>
        </key>
        <value>
          <int v="-35"/>
        </value>
        <key>
          <str>saved</str>
        </key>
        <value>
          <date v="2022-03-21"/>
        </value>
        <key>
          <str>volume</str>
        </key>
        <value>
          <real v="0.7"/>
        </value>
      </map>
    </value>
    <key>
      <str>Magic</str>
    </key>
    <value>
      <bytes>1F8B</bytes>
    </value>
    <key>
      <str>Nested Map</str>
    </key>
    <value>
      <map>
        <key>
          <str>Classical</str>
        </key>
        <value>
          <list>
            <int v="5"/>
            <yes/>
          </list>
        </value>
        <key>
          <str>Modern Instrumental</str>
        </key>
        <value>
          <list>
            <int v="4"/>
            <no/>
          </list>
        </value>
        <key>
          <str>Nested List of Lists</str>
        </key>
        <value>
          <list>
            <list>
              <int v="5"/>
              <str>Classical</str>
              <yes/>
            </list>
            <list>
              <int v="4"/>
              <str>Modern Instrumental</str>
              <no/>
            </list>
            <list>
              <int v="1"/>
              <str>New Acquistions</str>
              <no/>
            </list>
          </list>
        </value>
        <key>
          <str>Nested List of Tables</str>
        </key>
        <value>
          <list>
            <table name="Categories">
              <row>
                <int v="5"/>
                <str>Classical</str>
                <yes/>
              </row>
              <row>
                <int v="4"/>
                <str>Modern Instrumental</str>
                <no/>
              </row>
              <row>
                <int v="1"/>
                <str>New Acquistions</str>
                <no/>
              </row>
              <row>
                <int v="2"/>
                <str>Pop</str>
                <no/>
              </row>
              <row>
                <int v="3"/>
                <str>Punk</str>
                <no/>
              </row>
              <row>
                <int v="7"/>
                <str>Uncategorized</str>
                <no/>
              </row>
              <row>
                <int v="6"/>
                <str>Unpopular Pop</str>
                <no/>
              </row>
            </table>
            <table name="Playlists">
              <row>
                <int v="4"/>
                <str>ABBA</str>
                <int v="2"/>
                <no/>
              </row>
              <row>
                <int v="38"/>
                <str>Bach</str>
                <int v="5"/>
                <no/>
              </row>
              <row>
                <int v="39"/>
                <str>Bartok</str>
                <int v="5"/>
                <no/>
              </row>
              <row>
                <int v="5"/>
                <str>Beatles</str>
                <int v="2"/>
                <no/>
              </row>
              <row>
                <int v="40"/>
                <str>Beethoven</str>
                <int v="5"/>
                <no/>
              </row>
              <row>
                <int v="6"/>
                <str>Blondie</str>
                <int v="2"/>
                <no/>
              </row>
              <row>
                <int v="52"/>
                <str>Bob Marley</str>
                <int v="6"/>
                <yes/>
              </row>
              <row>
                <int v="7"/>
                <str>Bruce Springsteen</str>
                <int v="2"/>
                <no/>
              </row>
              <row>
                <int v="41"/>
                <str>Chopin</str>
                <int v="5"/>
                <yes/>
              </row>
              <row>
                <int v="37"/>
                <str>Classical</str>
                <int v="5"/>
                <no/>
              </row>
              <row>
                <int v="8"/>
                <str>David Bowie</str>
                <int v="2"/>
                <no/>
              </row>
              <row>
                <int v="9"/>
                <str>Dire Straits</str>
                <int v="2"/>
                <no/>
              </row>
            </table>
          </list>
        </value>
        <key>
          <str>Nested Table</str>
        </key>
        <value>
          <table name="Categories">
            <row>
              <int v="5"/>
              <str>Classical</str>
              <yes/>
            </row>
            <row>
              <int v="4"/>
              <str>Modern Instrumental</str>
              <no/>
            </row>
            <row>
              <int v="1"/>
              <str>New Acquistions</str>
              <no/>
            </row>
            <row>
              <int v="2"/>
              <str>Pop</str>
              <no/>
            </row>
            <row>
              <int v="3"/>
              <str>Punk</str>
              <no/>
            </row>
            <row>
              <int v="7"/>
              <str>Uncategorized</str>
              <no/>
            </row>
            <row>
              <int v="6"/>
              <str>Unpopular Pop</str>
              <no/>
            </row>
          </table>
        </value>
        <key>
          <str>New Acquistions</str>
        </key>
        <value>
          <list>
            <int v="1"/>
            <no/>
          </list>
        </value>
      </map>
    </value>
    <key>
      <str>Window</str>
    </key>
    <value>
      <map>
        <key>
          <str>height</str>
        </key>
        <value>
          <int v="536"/>
        </value>
        <key>
          <str>scale</str>
        </key>
        <value>
          <real v="1.1"/>
        </value>
        <key>
          <str>width</str>
        </key>
        <value>
          <int v="590"/>
        </value>
        <key>
          <str>x</str>
        </key>
        <value>
          <int v="383"/>
        </value>
        <key>
          <str>y</str>
        </key>
        <value>
          <int v="124"/>
        </value>
      </map>
    </value>
  </map>
</uxf>
